back/engines/commercial/granular_permissions/spec/acceptance/permissions_spec.rb

```diff
@@ -234,5 +263,107 @@
         })
       end
     end
+
+    get 'web_api/v1/permissions/:action/requirements' do
+      before do
+        @permission = Permission.where(permission_scope_type: nil).first
+        @permission.update!(permitted_by: 'everyone')
+      end
+
+      let(:action) { @permission.action }
+
+      example_request 'Get the participation requirements of a user in the global scope' do
+        assert_status 200
+        json_response = json_parse response_body
+        expect(json_response).to eq({
+          permitted: true,
+          requirements: {
+            built_in: {
+              first_name: 'satisfied',
+              last_name: 'satisfied',
+              email: 'satisfied'
+            },
+            custom_fields: {},
+            special: {
+              password: 'satisfied',
+              confirmation: 'satisfied'
+            }
+          }
+        })
+      end
+    end
+
+    get 'web_api/v1/permissions/:action/requirements' do
+      before do
+        create :custom_field_birthyear, required: true
+        create :custom_field_gender, required: false
+        create :custom_field_checkbox, resource_type: 'User', required: true, key: 'extra_field'
+
+        @user.update!(
+          email: 'my@email.com',
+          first_name: 'Jack',
+          last_name: nil,
+          password_digest: nil,
+          custom_field_values: { 'gender' => 'male' }
+        )
+      end
+
+      let(:action) { 'visiting' }
+
+      example_request 'Get the global registration requirements' do
```

Comment:
Not sure the difference between the description of this and the previous request test really tells me what's different. Second one maybe needs to be called, 'Get the global registration requirements when custom fields are asked'

Reply:
Sure! I changed the description.